### PR TITLE
Make core packages null by default instead of excluding them

### PR DESCRIPTION
**Pull request by shlevy**
This allows overriding core packages

### CODE REVIEW

**Comment by shlevy** [comment]
Note that before updating `nixpkgs` to use this you must set `pretty = null` in `haskell-packages.nix` otherwise the agda `pretty` package is used instead.

**Comment by shlevy** [comment]
@peti We have a somewhat pressing need to use this, I'd appreciate a quick review if possible.

File: src/Cabal2Nix/CorePackages.hs [hunk 1]
```diff
@@ -8,7 +10,7 @@ module Cabal2Nix.CorePackages ( corePackages, coreBuildTools ) where
 -- The commented packages have previously been non-core, so we don't filter
 -- them.
 corePackages :: [String]
-corePackages = [
+corePackages = map toNixName [
```
Comment (peti) [inline src/Cabal2Nix/CorePackages.hs]: What is the purpose of this change? Currently, `corePackages` contain a list of Haskell package names, *not* Nix attributes. Since the latter can be generated out of the former -- but not vice versa -- it seems like a bad idea to apply `toNixName` here.

File: src/Cabal2Nix/CorePackages.hs [hunk 2]
```diff
@@ -37,7 +39,7 @@ corePackages = [
   ]
 
 coreBuildTools :: [String]
-coreBuildTools = [
+coreBuildTools = concatMap buildToolNixName [
```
Comment (peti) [inline src/Cabal2Nix/CorePackages.hs]: Ditto.

File: src/Cabal2Nix/CorePackages.hs (hunk 1 above)
Comment (shlevy) [inline src/Cabal2Nix/CorePackages.hs]: The lists are only used in one place, both before and after this patch. But sure, I'll move this closer to where it's used (when I'm back at my computer)

**Comment by peti** [comment]
Are you aware of the fact that `corePackages` will be empty altogether in v2.0, https://github.com/NixOS/cabal2nix/commit/8a5ec344e72250b8ccdbb34a311148fcd0c23135?

**Comment by peti** [comment]
I'd rather remove the filtering of core packages altogether, like it's already done in the v2.x branch, and provide `null` overrides for all of them rather than relying on default null values in every expression.

File: src/Cabal2Nix/CorePackages.hs (hunk 1 above)
Comment (shlevy) [inline src/Cabal2Nix/CorePackages.hs]: Fixed this in force push

**Comment by shlevy** [comment]
No, I didn't know that. What's the time frame for switching to 2.0?

**Comment by peti** [comment]
Some time later today.

**Comment by shlevy** [comment]
Ah, cool! Perfect then, I'll close this.

Does `2.0` work on non-haskell-ng?

**Comment by peti** [comment]
The derivation format in NG is different; so 2.x cannot be used to maintain the old stuff.
